tests/system_tests_http2.py

```diff
@@ -447,6 +451,39 @@ def test_get_image_jpg(self):
         digest_of_response_file = get_digest(self.router_qdra.outdir + image_file_name)
         self.assertEqual(digest_of_server_file, digest_of_response_file)
 
+    def test_send_rst_stream_after_get_request(self):
+        # Send a GET request to get test.jpg (which is a large file) and immediately after the GET
+        # request send an RST_STREAM frame.
+        # There should be no http2 stream object leaks.
+        client_conn = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        client_conn.settimeout(TIMEOUT)
+        client_conn.connect(('127.0.0.1', self.http_listener_port))
+
+        # Send the HTTP2 raw bytes
+        # HTTP2 Client Magic
+        data = bytes.fromhex('505249202a20485454502f322e300d0a0d0a534d0d0a0d0a')
+        client_conn.send(data)
+
+        # HTTP2 Settings frame
+        data = bytes.fromhex('000012040000000000000300000064000402000000000200000000')
+        client_conn.send(data)
+        # HTTP2 Window Update frame
+        data = bytes.fromhex('00000408000000000001ff0001')
+        client_conn.send(data)
+
+        # HTTP2 Get request - /images/test.jpg
+        data = bytes.fromhex(
+            '00002b01050000000182048c60d48e62a18495095fa5737f86418aa0e41d139d09b8f8000f7a8825b650c3abbcdae053032a2f2a')
+        client_conn.send(data)
+
+        # Send the RST_STREAM frame
+        data = bytes.fromhex('00000403000000000100000008')
+        client_conn.send(data)
+
+        self.router_qdra.wait_log_message("HTTP2 NGHTTP2_RST_STREAM frame received, freeing stream data")
```

Comment:
I executed the following command to run the http2 tests - 
`[gmurthy@localhost build]$ ctest -V -R http2`
I am seeing that the test `test_send_rst_stream_after_get_request()` getting executed only once. 
Are you seeing the test executed more than once ? Thx

Reply:
I also checked in CI and did not see the test execute more than once. Are you guys seeing the test executing more than once in CI ? This is the only test that sends the http2 RST_STREAM, so, technically the log message must only appear once in the log.

Reply:
Why is github adding a smiling face emoji after every comment ?

Reply:
> I also checked in CI and did not see the test execute more than once. Are you guys seeing the test executing more than once in CI ? This is the only test that sends the http2 RST_STREAM, so, technically the log message must only appear once in the log.

That's what it is now. Ideally, tests should be written so you can reason about them in isolation, and results of one test cannot impact result of any other test. Of course, for _reasons_ (mostly performance) that's not how this is done, and e.g. tests share some setup, etc. but still, effort should be made so that the tests don't influence themself.

I can imagine putting this change (check logs starting from the point where particular test started) into backlog, and maybe not addressing it right now. But that's up to @kgiusti how much he is bothered by it right here.

> Why is github adding a smiling face emoji after every comment ?

@ganeshmurthy Try clicking it!

Reply:
> Try clicking it!

Very unintuitive to me.

Reply:
@ganeshmurthy oh, sorry - no I didn't mean the test was running twice.

What I meant is that the test will pass if it sees the log message in the router's logfile.  That may not work if other test_xxx routines in that class can trigger that log message as well.  Or if you add any further tests in the future to that class that trigger that message.

The test assumes the first matching log message it reads in the log was generated by that particular test. That may not be true if some other testcase "test_xxx() function" happens to trigger that log message before this test runs.

Jiri has some suggestions for working around this. Or you can cheat and name the test function "test_000_send_rst_stream_after_get_request" which would force the test to run first before any other tests in that class.

Reply:
I have set the test name to `test_000_send_rst_stream_after_get_request` for now